src/common/hooks/useRestoreToken.tsx

```
@@ -0,0 +1,96 @@
+  unknown: "0x4Bf7E4777a8D1b6EdD5F2d9b8582e2817F0B0953",
+};
+
+//Wraper Method to connect to instacne of TitleEscrowFactory
+export const getTitleEscrowCreator = async (provider: providers.Provider): Promise<TitleEscrowCreator> => {
+  const creatorContractAddress = CREATOR_CONTRACTS[NETWORK];
+  if (!creatorContractAddress) throw new Error(`Title escrow contract creator is not declared for ${NETWORK} network`);
+  return TitleEscrowCreatorFactory.connect(creatorContractAddress, provider);
+};
+
+/**
+ * This hook restores tokens that has been surrendered to Token Registry
+ * Restroing a document which has been surrendered to issuer takes 2 steps:
+ * 1. Deploying TitleEscrow with owner and holder from last known beneficary
+ * 2. Transfering ownership of document from Token registry to Titleescrow in step 1
+ * @returns function which combines from process decribed
+ * errorMessage is populated if any other error is returned
+ * state is changed based on the step of the ethereum transaction
+ */
+export const useRestoreToken = (
+  contractInstance: TradeTrustErc721 | undefined,
+  provider: providers.Provider | Signer,
+  tokenId: string | undefined
+) => {
+  const [errorMessage, setErrorMessage] = useState<string>();
+  const [state, setState] = useState<ContractFunctionState>("UNINITIALIZED");
+
+  const { send: sendToken, reset: sendTokenResetState } = useContractFunctionHook(contractInstance, "sendToken");
+
+  const restoreToken = async (previousBeneficary: string): Promise<void> => {
+    setState("INITIALIZED");
+    try {
+      if (!tokenId) throw new Error("Ownership data is not provided");
+      const titleEscrowCreatorContract = await getTitleEscrowCreator(provider as providers.Provider);
+
+      // Deploy new title escrow smart contract to own document
+      const escrowDeploymentReceipt = await titleEscrowCreatorContract.deployNewTitleEscrow(
+        contractInstance?.address ?? "",
```

This should never be "" right? Should throw an error right after try statement to assert that first.